Graduate SeccompDefault feature to beta (#34640)
kubernetes/website: content/en/docs/tutorials/security/seccomp.md

kind documentation about configuration for more details on this.
This tutorial assumes you are using Kubernetes {{< param "version" >}}.

> As an beta feature, you can configure Kubernetes to use the profile that the

Author: Thanks, fixed!

## Enable the use of `RuntimeDefault` as the default seccomp profile for all workloads

{{< feature-state state="beta" for_k8s_version="v1.25" >}}

To use seccomp profile defaulting, you must run the kubelet with the `SeccompDefault`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled
(this is the default). You must also explicitly enable the defaulting behavior for each

> node where you want to use this.
> well as corresponding `--seccomp-default`

Reviewer: ```suggestion
node where you want to use this with the corresponding `--seccomp-default`
```

Author: Good catch, thank you!

## Enable the use of `RuntimeDefault` as the default seccomp profile for all workloads

{{< feature-state state="beta" for_k8s_version="v1.25" >}}

> `SeccompDefault` is an optional kubelet
> [feature gate](/docs/reference/command-line-tools-reference/feature-gates) as

Reviewer 2: ```suggestion
To use seccomp profile defaulting, you must run the kubelet with the `SeccompDefault`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled
(this is the default). You must also explicitly enable the defaulting behavior for each
node where you want to use this.
```
**(and then revise the next line).**

recommends that you enable this feature gate on a subset of your nodes and then
test workload execution before rolling the change out cluster-wide.

You can find more detailed information about a possible upgrade and downgrade strategy
in the related Kubernetes Enhancement Proposal (KEP):
[Enable seccomp by default](https://github.com/kubernetes/enhancements/tree/9a124fd29d1f9ddf2ff455c49a630e3181992c25/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).

> Seccomp defaulting for Pods is a beta feature in Kubernetes {{< skew currentVersion >}}, 

Reviewer 3: Question: Is the feature called 'Seccomp defaulting' or 'Seccomp'?
I'm not sure I understand the use of the word, 'defaulting'. Which or what is 'defaulting'?
`However, you still need to enable this defaulting for each node where
you would like to use it.`

Author: Right, fixing it in https://github.com/kubernetes/website/pull/35121

recommends that you enable this feature gate on a subset of your nodes and then
test workload execution before rolling the change out cluster-wide.

You can find more detailed information about a possible upgrade and downgrade strategy
in the related Kubernetes Enhancement Proposal (KEP):
[Enable seccomp by default](https://github.com/kubernetes/enhancements/tree/9a124fd29d1f9ddf2ff455c49a630e3181992c25/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).

Seccomp defaulting for Pods is a beta feature in Kubernetes {{< skew currentVersion >}}, 
and the corresponding `SeccompDefault` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
is enabled by default. However, you still need to enable this defaulting for each node where
you would like to use it.

> If you are running Kubernetes {{< skew currentVersion >}} cluster and want to enable Seccomp

Reviewer 2: Oops.
```suggestion
If you are running Kubernetes a {{< skew currentVersion >}} cluster and want to enable Seccomp
```

> More detailed information about a possible upgrade and downgrade strategy can be
> found in the [related Kubernetes Enhancement Proposal (KEP)](https://github.com/kubernetes/enhancements/tree/a70cc18/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).

Reviewer 2: ```suggestion
You can find more detailed information about a possible upgrade and downgrade strategy
in the related Kubernetes Enhancement Proposal (KEP):
[Enable seccomp by default](https://github.com/kubernetes/enhancements/tree/9a124fd29d1f9ddf2ff455c49a630e3181992c25/keps/sig-node/2413-seccomp-by-default#upgrade--downgrade-strategy).
```
Notice that I bumped to commit https://github.com/kubernetes/enhancements/commit/9a124fd29d1f9ddf2ff455c49a630e3181992c25

> `SeccompDefault` is in beta state, which means that the corresponding feature
> gate is `true` by default. To additionally enable the feature, pass the
> flag `--seccomp-default` to the `kubelet` CLI or enable it via the [kubelet

Reviewer 2: ```suggestion
Seccomp defaulting for Pods is a beta feature in Kubernetes {{< skew currentVersion >}}, 
and the corresponding `SeccompDefault` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
is enabled by default. However, you still need to enable this defaulting for each node where
you would like to use it.

If you are running Kubernetes {{< skew currentVersion >}} cluster and want to enable Seccomp
defaulting, either run the kubelet with the `--seccomp-default` command line flag, or enable
Seccomp defaulting through the
[kubelet
```
